File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/AdverseEvent.ts
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type AdverseEventSuspectEntityCausality = BackboneElement & {
    author?: Reference;
    method?: CodeableConcept;
    assessment?: CodeableConcept;
    productRelatedness?: string;
};

export type AdverseEventSuspectEntity = BackboneElement & {
    instance: Reference;
    causality?: AdverseEventSuspectEntityCausality[];
};

export type AdverseEvent = DomainResource & {
    category?: CodeableConcept[];
    actuality: string;
    date?: string;
    study?: Reference[];
    encounter?: Reference;
    suspectEntity?: AdverseEventSuspectEntity[];
    referenceDocument?: Reference[];
    outcome?: CodeableConcept;
    recordedDate?: string;
    event?: CodeableConcept;
    contributor?: Reference[];
    subjectMedicalHistory?: Reference[];
    recorder?: Reference;
    seriousness?: CodeableConcept;
    severity?: CodeableConcept;
    identifier?: Identifier;
    detected?: string;
    location?: Reference;
    subject: Reference;
    resultingCondition?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/AdverseEvent.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class AdverseEventSuspectEntityCausality(BackboneElement):
    author: Optional[Reference] = None
    method: Optional[CodeableConcept] = None
    assessment: Optional[CodeableConcept] = None
    product_relatedness: Optional[str] = None

@dataclass(kw_only=True)
class AdverseEventSuspectEntity(BackboneElement):
    instance: Reference
    causality: Optional[List[AdverseEventSuspectEntityCausality]] = field(default_factory=list)

@dataclass(kw_only=True)
class AdverseEvent(DomainResource):
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    actuality: str
    date: Optional[str] = None
    study: Optional[List[Reference]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    suspect_entity: Optional[List[AdverseEventSuspectEntity]] = field(default_factory=list)
    reference_document: Optional[List[Reference]] = field(default_factory=list)
    outcome: Optional[CodeableConcept] = None
    recorded_date: Optional[str] = None
    event: Optional[CodeableConcept] = None
    contributor: Optional[List[Reference]] = field(default_factory=list)
    subject_medical_history: Optional[List[Reference]] = field(default_factory=list)
    recorder: Optional[Reference] = None
    seriousness: Optional[CodeableConcept] = None
    severity: Optional[CodeableConcept] = None
    identifier: Optional[Identifier] = None
    detected: Optional[str] = None
    location: Optional[Reference] = None
    subject: Reference
    resulting_condition: Optional[List[Reference]] = field(default_factory=list)